xapian-core/weight/tfidfweight.cc

```diff
@@ -70,26 +74,48 @@ TfIdfWeight::name() const
 string
 TfIdfWeight::serialise() const
 {
-    return normalizations;
+    if (param_s > 0 || param_delta > 0) {
+	string result = serialise_double(param_s);
+	result += serialise_double(param_delta);
+	return result;
+    } else {
+	return normalizations;
+    }
 }
 
 TfIdfWeight *
-TfIdfWeight::unserialise(const string & s) const
+TfIdfWeight::unserialise(const string & str) const
 {
-    if (s.length() != 3)
-	throw Xapian::SerialisationError("Extra data in TfIdfWeight::unserialise()");
-    return new TfIdfWeight(s);
+    if (param_s > 0 || param_delta > 0) {
```

Comment:
We don't need to worry about setting parameters to zero as they have been purposely initialised to zero. And since we want only non-negative and non-zero values of parameters, this makes sure appropriate serialisation takes place when constructing `TfIdfWeight` either for normalization strings or passing parameter values. This [test](https://github.com/xapian/xapian/pull/115/files?diff=unified#diff-89288dcbe247bb4c55a8f4591ebda81bR672) demonstrates the idea.

And, this [test ](https://github.com/xapian/xapian/pull/115/files?diff=unified#diff-89288dcbe247bb4c55a8f4591ebda81bR688) is an example to see how we'll be using pivoted normalization where one constructor (at line 690) to select `"PPP"` normalization string and another in the next line to pass parameter values.